bgpd/bgp_flowspec_vty.c

```diff
@@ -332,16 +332,32 @@ void route_vty_out_flowspec(struct vty *vty, struct prefix *p,
 		struct bgp_info_extra *extra = bgp_info_extra_get(binfo);
 
 		if (extra->bgp_fs_pbr) {
+			struct listnode *node;
 			struct bgp_pbr_match_entry *bpme;
 			struct bgp_pbr_match *bpm;
+			int unit = 0;
+			struct list *list_bpm;
 
-			bpme = (struct bgp_pbr_match_entry *)extra->bgp_fs_pbr;
-			bpm = bpme->backpointer;
-			vty_out(vty, "\tinstalled in PBR");
-			if (bpm)
-				vty_out(vty, " (%s)\n", bpm->ipset_name);
-			else
-				vty_out(vty, "\n");
+			list_bpm = list_new();
+			if (listcount(extra->bgp_fs_pbr))
+				vty_out(vty, "\tinstalled in PBR");
+			for (ALL_LIST_ELEMENTS_RO(extra->bgp_fs_pbr,
+						  node, bpme)) {
+				bpm = bpme->backpointer;
+				if (listnode_lookup(list_bpm, bpm))
```

Comment:
is a non-sorted linked list the appropriate data structure to store data we have already displayed?  How many items can be in this list?

Reply:
a FS entry may be made up of several or combinations. ( as many as the FS size, I think 4096 bytes). Each combination stands for a PBR entry.



Theorically, this can be several hundred
Practically, 10 is an example on the situations I faced.

most of the time 1 FS = 1 PBR entry. 

FS entries are ordered in hash lists.
I admit having a hash list would be nice. This could be a separate work.
Right now, this list is only used for display.

Is it ok to go ahead with that status ?

Reply:
yeah it's fine.  I was just trying to understand when this is going to become a problematic data structure.